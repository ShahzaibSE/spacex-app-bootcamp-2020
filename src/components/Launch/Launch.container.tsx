import React from 'react'

export const LaunchContainerComponent = () => {
    return (
        <div>
            
        </div>
    )
}
