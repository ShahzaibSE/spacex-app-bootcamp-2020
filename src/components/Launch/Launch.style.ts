import { makeStyles } from '@material-ui/core/styles';

export const launchCardStyles = makeStyles({
    root: {
      maxWidth: 345,
      marginBottom:20,
      marginTop:20,
      marginLeft:"10%",
      marginRight:"10%"
    },
    media: {
      height: 200,
    },
});