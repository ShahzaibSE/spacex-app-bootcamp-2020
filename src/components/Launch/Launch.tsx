import React, {FC} from 'react'
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import {LaunchesQuery} from "./../../generated/graphql"
// Assets.
import {launchCardStyles} from "./Launch.style"
import rocket from "../../static/images/rocket.jpg"

type LaunchProps = {
  data: LaunchesQuery
}

const Launch: FC<LaunchProps> = ({data}: any) => {
    let launch = data
    let launch_video_link:string = data.links.video_link
    const classes = launchCardStyles();
    console.log("Single Launch Information")
    console.log(data)

    // Creating Video link.
    const create_video_link = (video_id:string) => {
      let hostname = "youtube"
      let video_link = `https://www.${hostname}.com/embed/${video_id}`
      console.log(`Created video link: ${video_link}`)
      return video_link
    }

    if (launch_video_link == null){
      console.log("Video link not found")
      launch_video_link = rocket
    } else {
      console.log("Video ID extracted from video link")
      console.log(launch_video_link.split(/[\s/v=]+/))
      launch_video_link = create_video_link(launch_video_link.split(/[\s/v=]+/)
                                            [launch_video_link.split(/[\s/v=]+/).length-1])
    }

    return (
        <Card className={classes.root}>
        <CardActionArea>
          <CardMedia component="iframe"
            className={classes.media}
            // src={"https://www.youtube.com/watch?v=JuZBOUMsYws".replace("watch?v=", "embed/")}
            src={launch_video_link}
            title="SpaceX Launch Mission"
          />
          <CardContent>
            <Typography gutterBottom variant="h5" component="h2">
              Mission: Falcon
            </Typography>
            <Typography variant="body2" color="textSecondary" component="p">
              Lizards are a widespread group of squamate reptiles, with over 6,000 species, ranging
              across all continents except Antarctica
            </Typography>
          </CardContent>
        </CardActionArea>
        <CardActions>
          <Button size="small" color="primary">
            Learn More
          </Button>
        </CardActions>
      </Card>
    )
}

export default Launch
