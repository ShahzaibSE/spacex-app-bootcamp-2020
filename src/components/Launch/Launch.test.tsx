import React from "react";
import {shallow, mount} from "enzyme";
// Component.
import Launch from "./Launch";

describe("<Launch/> tests", ()=>{
    let container:any;
    let dummy_data:any = {flight_number:1, mission_name:"Falcon", links:{video_link:""}}

    beforeEach(()=>{
        container = shallow(<Launch data={dummy_data}/>)
    })

    it("check if <Launch/> card has rendered",()=>{
        expect(container).toMatchSnapshot()
    })
})