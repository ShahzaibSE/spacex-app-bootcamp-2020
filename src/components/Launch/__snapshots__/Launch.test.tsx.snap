// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Launch/> tests check if <Launch/> card has rendered 1`] = `
<div>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-root-1"
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(CardActionArea))>
      <WithStyles(ForwardRef(CardMedia))
        className="makeStyles-media-2"
        component="iframe"
        src="https://www.youtube.com/embed/"
        title="Falcon"
      />
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h2"
          gutterBottom={true}
          variant="h5"
        >
          Mission: 
          Falcon
           (
          )
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          component="h2"
          gutterBottom={true}
          style={
            Object {
              "color": "#9e9e9e",
            }
          }
          variant="subtitle1"
        />
      </WithStyles(ForwardRef(CardContent))>
    </WithStyles(ForwardRef(CardActionArea))>
    <WithStyles(ForwardRef(CardActions))
      className="makeStyles-action_container-3"
    >
      <WithStyles(ForwardRef(Button))
        disabled={true}
        size="small"
        variant="contained"
      >
        N/A
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(CardActions))>
  </WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(Dialog))
    TransitionComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    fullScreen={true}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(AppBar))
      className="makeStyles-appBar-5"
    >
      <WithStyles(ForwardRef(Toolbar))>
        <WithStyles(ForwardRef(IconButton))
          aria-label="close"
          color="inherit"
          edge="start"
          onClick={[Function]}
        >
          <CloseIcon />
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-title-6"
          variant="h6"
        >
          Falcon
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Toolbar))>
    </WithStyles(ForwardRef(AppBar))>
    <LaunchDetailsContainerComponent
      launch={
        Object {
          "flight_number": 1,
          "links": Object {
            "video_link": "",
          },
          "mission_name": "Falcon",
        }
      }
    />
  </WithStyles(ForwardRef(Dialog))>
</div>
`;
